src/Support/Mutex.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+namespace RonasIT\AutoDoc\Support;
+
+use Illuminate\Support\Arr;
+use RuntimeException;
+
+class Mutex
+{
+    protected array $config;
+
+    protected const string MODE_FILE_READ_WRITE = 'c+';
+    protected const string MODE_FILE_READ = 'r';
+
+    public function __construct()
+    {
+        $this->config = config('auto-doc.paratests');
+    }
+
+    public function readFileWithLock(string $filePath): ?string
+    {
+        $handle = fopen($filePath, self::MODE_FILE_READ);
+
+        try {
+            $this->acquireLock($handle, LOCK_SH);
+
+            $content = stream_get_contents($handle);
+
+            return ($content === false) ? null : $content;
```

Comment:
could we do it like this?

```suggestion
            return (string) stream_get_contents($handle);
```